### PR TITLE
Adding wls plugin

### PR DESCRIPTION
**Pull request by mohatb**

### CODE REVIEW

File: plugins/wls.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+    name: wls
+spec:
+    version: v1.0.0
+    homepage: 'https://github.com/mohatb/kubectl-wls'
+    shortDescription: 'Windows-linux-shell allows you to start a shell on windows or linux nodes'
```
Comment (ahmetb) [inline plugins/wls.yaml]: I don't understand anything from this sentence.

File: plugins/wls.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    description: "kubectl-wls stands for kubectl Kubectl-Windows-Linux-Shell, it is an interactive shell script and also a kubectl plugin that allows you to access a kubernetes windows or linux nodes shell.\n"
```
Comment (ahmetb) [inline plugins/wls.yaml]: First, please refer to other plugin manifest to see how to write multi-line strings that are wrapped at 80 chars.

Then, I don't understand what the plugin does from this sentence either.

> kubectl-wls stands for kubectl Kubectl-Windows-Linux-Shell

not useful

> it is an interactive shell script and also a kubectl plugin

sounds redundant

> allows you to access a kubernetes windows or linux nodes shell

what is a "kubernetes windows"? "linux nodes"?

Please use improved grammar.

**Comment by ahmetb** [comment]
Tests are failing.

Please read https://krew.sigs.k8s.io/docs/developer-guide/release/new-plugin/
and https://krew.sigs.k8s.io/docs/developer-guide/testing-locally/

File: plugins/wls.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    platforms:
+        -
+            selector: {matchExpressions: [{key: os, operator: In, values: [darwin, linux]}]}
+            uri: 'https://github.com/mohatb/kubectl-wls/archive/v1.0.0.tar.gz'
+            sha256: 1766ed7bb18dcf38cfe0a136246d731d94776b7292b632a668fba7bfb21b3f22
+            files: 
+              - from: '*'
+                to: "."
```
Comment (ahmetb) [inline plugins/wls.yaml]: this is optional, you can omit if you're copying out everything

File: plugins/wls.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    name: wls
+spec:
+    version: v1.0.0
+    homepage: 'https://github.com/mohatb/kubectl-wls'
+    shortDescription: 'Windows-linux-shell allows you to start a shell on windows or linux nodes'
+    description: "kubectl-wls stands for kubectl Kubectl-Windows-Linux-Shell, it is an interactive shell script and also a kubectl plugin that allows you to access a kubernetes windows or linux nodes shell.\n"
+    platforms:
+        -
+            selector: {matchExpressions: [{key: os, operator: In, values: [darwin, linux]}]}
+            uri: 'https://github.com/mohatb/kubectl-wls/archive/v1.0.0.tar.gz'
```
Comment (ahmetb) [inline plugins/wls.yaml]: your source code says

> $(kubectl get node aksnpwin000000

have you actually tested this thoroughly?

File: plugins/wls.yaml (hunk 1 above)
Comment (mohatb) [reply plugins/wls.yaml]: > wls: stands for windows-linux-shell 
> it allows the user to get into a windows or linux shell through this plugin, for linux it uses nsenter, for windows, it creates a pod on a linux node and then ssh to a windows node.

File: plugins/wls.yaml (hunk 4 above)
Comment (mohatb) [reply plugins/wls.yaml]: updated source with a variable instead

File: plugins/wls.yaml (hunk 1 above)
Comment (mohatb) [reply plugins/wls.yaml]: can you test the yaml and the plugin again ?

File: plugins/wls.yaml (hunk 4 above)
Comment (mohatb) [reply plugins/wls.yaml]: yes, and I updated both yaml and the source

File: plugins/wls.yaml (hunk 2 above)
Comment (mohatb) [reply plugins/wls.yaml]: in kubernetes, there is a windows nodes and linux nodes, this allows the user to ssh to any of those nodes. this does not need a grammer expert

Comment (mohatb) [reply plugins/wls.yaml]: in kubernetes, there are windows or Linux nodes, this supports accessing shell in either windows or Linux nodes.

File: plugins/wls.yaml (hunk 3 above)
Comment (mohatb) [reply plugins/wls.yaml]: updated the yaml with the required files.

File: plugins/wls.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: wls
+spec:
+  version: "v1.0.0"
+  homepage: https://github.com/mohatb/kubectl-wls
+  shortDescription: Windows-linux-shell allows you to start a shell on windows or linux nodes
```
Comment (ahmetb) [inline plugins/wls.yaml]: `Windows-linux-shell allows you to ` is redundant, makes "short" description too long.

File: plugins/wls.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  description: |
+    kubectl-wls it is an interactive shell script that allows you to access a kubernetes windows or linux node shell.
```
Comment (ahmetb) [inline plugins/wls.yaml]: wrap at 80 chars

File: plugins/wls.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      - key: os
+        operator: In
+        values:
+        - darwin
+        - linux
+    uri: https://github.com/mohatb/kubectl-wls/archive/v1.0.0.tar.gz
+    sha256: "43e571371da3ae71ed804c7baf78f1ddcc35127e3375d815245034511d8913b5"
+    files:
+    - from: "kubectl-wls*"
+      to: "kubectl-wls"
```
Comment (ahmetb) [inline plugins/wls.yaml]: please copy out LICENSE

Comment (ahmetb) [reply plugins/wls.yaml]: read the checklist at https://krew.sigs.k8s.io/docs/developer-guide/release/new-plugin/

**Comment by ahmetb** [comment]
Do you mind explaining how's this different than

- https://github.com/kubernetes-sigs/krew-index/blob/master/plugins/node-shell.yaml
- https://github.com/kubernetes-sigs/krew-index/blob/master/plugins/cssh.yaml
- https://github.com/kubernetes-sigs/krew-index/blob/master/plugins/ssh-jump.yaml

Maybe the "windows" part is a differentiator. What does it do/how does it work? I'm not seeing much windows related logic in the plugin script. Care to explain further?

**Pull request closed without merging**